### PR TITLE
Updated license from MIT to BSD 3

### PR DESCRIPTION
**Pull request by nickc1**
This closes issue #11

### CODE REVIEW

**Comment by takluyver** [comment]
I'm happy to change it for the sake of consistency, but can you also update the license classifier here:
https://github.com/jupyter/testpath/blob/bc3ea3c9532988fea63bb1c204f8ff835ddaa709/flit.ini#L9

You can see the list of valid classifiers here:
https://pypi.python.org/pypi?%3Aaction=list_classifiers

**Comment by Carreau** [comment]
I guess that technically we also need to ask all contributor if they are ok with changing License.

**Comment by takluyver** [comment]
I'm not sure. However, there's only a handful of non-me contributors, so let's be on the safe side:

@EdwardBetts @nehaljwani @QuLogic @adamchainz the proposal is to relicense testpath from MIT to BSD for consistency with other Jupyter projects. These two licenses are conceptually similar, and you can see the details in the diff section of this PR. Are you all OK with that?

If we don't hear from you within two weeks, we'll assume you don't mind.

**Comment by nehaljwani** [comment]
I have no objections in switching to the BSD 3-Clause License.

**Comment by EdwardBetts** [comment]
Fine with me.

File: LICENSE [hunk 1]
```diff
@@ -1,21 +1,29 @@
-The MIT License (MIT)
+BSD 3-Clause License
 
-Copyright (c) 2015, The Jupyter Development Team
+Copyright (c) 2017, Project Jupyter Contributors
```
Comment (Carreau) [inline LICENSE]: Also please don't change the year. A single year is implicitely YYYY-Present, changing the year would basically be saying we relinquish our copyright for the 2015-2017 period.

**Comment by adamchainz** [comment]
Fine by me.

**Comment by QuLogic** [comment]
Fine with me as well.

**Comment by takluyver** [comment]
Thanks everyone.

**Pull request merged**